### PR TITLE
[FIX] web_editor: fix link regex

### PR DESCRIPTION
**Pull request by Goaman**
The regex that automatically detect link when pasting text or
when pressing space when editing was not good enough and
for instance did not capture fully an url like:
https://1234-abc.runbot007.odoo.com/web#id=3&menu_id=221

Task ID: 2602725

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/lib/odoo-editor/test/spec/urlRegex.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { URL_REGEX } from '../../src/OdooEditor.js';
+
+describe('urlRegex', () => {
```
Comment (Zinston) [inline addons/web_editor/static/lib/odoo-editor/test/spec/urlRegex.test.js:3]: :+1:

**Review by Zynton — CHANGES_REQUESTED** [review]
Commit looks good to me. Thanks for having added tests ;-)
I made the mistake of letting the previous one slide but you should follow the guidelines regarding commit messages:

https://www.odoo.com/documentation/master/developer/misc/other/guidelines.html?highlight=guidelines#commit-message-header
> It should be self explanatory and include the reason behind the change. Do not use single words like “bugfix” or “improvements”.

**Comment by Goaman** [comment]
> Commit looks good to me. Thanks for having added tests ;-)
> I made the mistake of letting the previous one slide but you should follow the guidelines regarding commit messages:
> 
> https://www.odoo.com/documentation/master/developer/misc/other/guidelines.html?highlight=guidelines#commit-message-header
> 
> > It should be self explanatory and include the reason behind the change. Do not use single words like “bugfix” or “improvements”.

I think it is self explanatory, the reason is also implicit, the regex is broken so we fix it.
This commit header is not a single word.

If you have a better message, can you make a suggestion ?

**Comment by Zinston** [comment]
@Goaman 
> If you have a better message, can you make a suggestion ?

`[FIX] web_editor: properly capture any form of url on paste` for example. This tells _what behavior is fixed_ rather than just the code you touched. That part is for the commit message if even needed. The header should give a good idea of the _purpose_ of the commit for someone stumbling on it, who has no context and no particular knowledge of the workings of the editor.
Think of it this way: someone stumbles on the regex and blames it to see why it looks the way it does. Then they see "fix the regex". No information was learned, now they have to waste time looking into the history. Whereas if they see "properly capture any form of url on paste" they learned that it had something to do with pasting and being less restrictive with the url forms that could be captured. They saved time, energy and frustration.

**Comment by Goaman** [comment]
Done.

**Comment by Zinston** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**